### PR TITLE
[#3746] improvement(spark-connector): Add scala version to maven artifact name for spark-connector

### PR DESCRIPTION
**Pull request by yuqi1129**
### What changes were proposed in this pull request?

Add scala version to the name of maven artifact. 

### Why are the changes needed?

Currently, we support multiple versions of scala version, if we don't include the scale version information, users would be unable to use jars with different scala version.

Fix: #3746

### Does this PR introduce _any_ user-facing change?

N/A.

### How was this patch tested?

image

### CODE REVIEW

File: spark-connector/v3.4/spark-runtime/build.gradle.kts [hunk 1]
```diff
@@ -13,7 +13,7 @@ plugins {
 val scalaVersion: String = project.properties["scalaVersion"] as? String ?: extra["defaultScalaVersion"].toString()
 val sparkVersion: String = libs.versions.spark34.get()
 val sparkMajorVersion: String = sparkVersion.substringBeforeLast(".")
-val baseName = "${rootProject.name}-spark-connector-runtime-${sparkMajorVersion}_$scalaVersion"
```
Comment (FANNG1) [inline spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: why remove `${rootProject.name}`?

Comment (FANNG1) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: gravitino-spark-connector-runtime seems more meaningful

Comment (yuqi1129) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: The artifact name for other jars does not contains `gravitino`

File: spark-connector/v3.4/spark/build.gradle.kts [hunk 2]
```diff
@@ -135,6 +135,18 @@ tasks.test {
   }
 }
```
Comment (FANNG1) [inline spark-connector/v3.4/spark/build.gradle.kts:137]: could you use a variable to represent `spark-${sparkMajorVersion}_${scalaVersion}`?

File: spark-connector/v3.4/spark-runtime/build.gradle.kts (hunk 1 above)
Comment (FANNG1) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: I perfer to keep `gravitino`,  `spark-connector-runtime` is too general to use in Spark

Comment (yuqi1129) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: > gravitino-spark-connector-runtime seems more meaningful

So, do we need to add prefix `gravitino` for other project?
image

File: spark-connector/v3.4/spark/build.gradle.kts (hunk 2 above)
Comment (yuqi1129) [reply spark-connector/v3.4/spark/build.gradle.kts:137]: OK

File: spark-connector/v3.4/spark-runtime/build.gradle.kts (hunk 1 above)
Comment (FANNG1) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: Does  artifact name must consistent with the package name? If yes,  I prefer to add `gravitino` to spark connector , if not, I'm ok to the artifact name.

Comment (FANNG1) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: no need to keep consistent with other modules like API?

Comment (jerqi) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: This is package will be used for Spark. So it's better to have project like iceberg jar.

Comment (yuqi1129) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: > Does artifact name must consistent with the package name?

I'm not 100% sure about it, but artifact names and package names are identical are the default behaviour.

Comment (FANNG1) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: ok, how about keep gravitino for spark connector, and keep the other modules unchanged?

Comment (yuqi1129) [reply spark-connector/v3.4/spark-runtime/build.gradle.kts:16]: I have not preference. If adding a gravitino prefix is more meaningful, I'm fine with it.

**Review by FANNG1 — APPROVED** [review]

**Pull request merged**